src/inc/quic_platform_winuser.h

```diff
@@ -763,9 +768,17 @@ CxPlatCqeUserData(
     _In_ const CXPLAT_CQE* cqe
     )
 {
-    return (void*)cqe->lpCompletionKey;
```

Comment:
Using the IOCP completion key to store user data is problematic because it prohibits providing unique userdata per IO, and so the existing model in Windows user is both inconsistent and inflexible. I've passed data using the overlapped struct, which is the common Windows idiom.

Reply:
But how does this compare with other platforms IO completion models? I'm not necessarily against this, but we didn't necessarily have a problem before.

Reply:
All other platforms have a single user_data macro for IO completion, which is per-SQE, not per-handle. The exception is the Windows kernel implementation, which is incomplete. It is a clear violation of the SQE->CQE pattern if the CQE is not derived from the IO's SQE, but instead derived from a separate, semi-global SQE.

In my humble opinion, we definitely had problems above - not in terms of functional issues with code crashing before, but in terms of quality of design and abstraction.

Reply:
What?! You're saying my code wasn't absolute perfection!? 😄

I'm Ok with these changes, but wish we didn't have to burn the extra pointer the change requires...